doc/man3/X509_digest.pod

```diff
@@ -39,21 +39,26 @@ PKCS7_ISSUER_AND_SERIAL_digest
 
 =head1 DESCRIPTION
 
-X509_digest_sig() calculates a digest of the given certificate
-using the same hash algorithm as in its signature with a fallback to B<SHA-256>
-for algorithms where the digest is an integral part of the signature algorithm
-such as with B<EdDSA> signatures.
+X509_digest_sig() calculates a digest of the given certificate I<cert>
+using the same hash algorithm as in its signature, if the digest
+is an integral part of the certificate signature algorithm identifier.
+In this case the I<default_md> argument is ignored.
```

Comment:
I'd suggest a slightly different semantics. That is to name the parameter `md_used` and always set it to the used md if it is non-NULL.

Reply:
This would be more consistent with other calls...

Reply:
I thought about this alternative as well, but at least for use with the upcoming CMPv3 
the caller needs to know whether the default was used or not, and only needs to know the chosen value in that case.
What we also could do is to have two output parameters: `md_used` and, e.g., `by_default`.

Reply:
I would suggest then to make the second parameter `int *md_is_fallback` or something in this sense.

Reply:
Another option would be to make the `default_md` an input parameter and have `int *default_used` as an output parameter. That makes the function slightly less useful though in general because for protocols where the hash algorithm would have to be specified in the message you would not know what the md is in the non-default case.

Reply:
> I would suggest then to make the second parameter `int *md_is_fallback` or something in this sense.

Done this way, since it appears to be the most generally useful solution.
(If the hash alg to be used is known beforehand, anyway `X509_digest()` can be used).